util/collate/collate.go

```diff
@@ -134,31 +134,28 @@ func CollationID2Name(id int32) string {
 type binPaddingCollator struct {
 }
 
-func (bpc *binPaddingCollator) Compare(a, b string, opt CollatorOption) int {
-	aLen := len(a)
-	bLen := len(b)
-	noPaddingResult := 0
-	if aLen > bLen {
-		noPaddingResult = strings.Compare(a[:bLen], b)
-		if noPaddingResult != 0 {
-			return noPaddingResult
-		}
-		return strings.Compare(a[bLen:], strings.Repeat(" ", aLen-bLen))
-	} else if aLen < bLen {
-		noPaddingResult = strings.Compare(a, b[:aLen])
-		if noPaddingResult != 0 {
-			return noPaddingResult
+func truncateTailingSpaceTwo(a, b string) (string, string) {
+	return truncateTailingSpaceOne(a), truncateTailingSpaceOne(b)
+}
+
+func truncateTailingSpaceOne(str string) string {
```

Comment:
What's the difference between this one and `truncateTrailingSpaces`?

Reply:
`truncateTrailingSpaces`  accepts `datum` and do some checks.

Reply:
Sorry, forgot to delete the latter.